Orange/widgets/visualize/owmosaic.py

```diff
@@ -152,7 +152,6 @@ def state_count(self):
         Return the number of combinations, starting with a single attribute
         if Mosaic is colored by class distributions, and two if by Pearson
         """
-        self.compute_attr_order()
```

Comment:
I agree it was not very good to compute the order every time one needed the count.
But that line was there because of the next one, which sets `n_attrs` using `attr_ordering`. Removing just the first and still relying on `len(self.attr_ordering)` is not the best.
Can you change the next line to actually compute the number of attributes? Should be trivial using len of the domain, just be careful to (not) count the class appropriately.
Then the extra calls in tests will not be needed as well.